src/flows/context/schemaProvider.tsx

```diff
@@ -69,22 +127,40 @@ export const SchemaProvider: FC<Props> = React.memo(({children}) => {
   }, [data?.bucket?.id, lastBucket, update])
 
   useEffect(() => {
-    dispatch(startWatchDog())
-  }, [dispatch])
-
-  useEffect(() => {
-    if (!data.bucket) {
+    if (!data?.bucket) {
       return
     }
 
-    dispatch(getAndSetBucketSchema(data.bucket, flow.range))
-  }, [data?.bucket, dispatch, flow?.range])
-
-  const schema = useSelector(
-    (state: AppState) => state.flow.schema[data.bucket?.name]?.schema || {}
-  )
+    setLoading(RemoteDataState.Loading)
+
+    const range = formatTimeRangeArguments(flow?.range)
+
+    const text = `from(bucket: "${data.bucket.name}")
+|> range(${range})
+|> first()
+|> drop(columns: ["_value"])
+|> group()`
+
+    const result = query(text)
+    result
+      .then((response: FluxResult) => {
+        const schemaForBucket = parsedResultToSchema(response.parsed)
+        setSchema(schemaForBucket)
+        setLoading(RemoteDataState.Done)
+      })
+      .catch((error: RunQueryResult) => {
+        console.error('error: ', error)
+        setLoading(RemoteDataState.Error)
+        setSchema({})
+      })
+  }, [data?.bucket?.name, data?.bucket, flow?.range, query])
 
-  const {fields, measurements, tags} = normalizeSchema(schema, data, searchTerm)
+  useEffect(() => {
+    const normalized = normalizeSchema(schema, data, searchTerm)
```

Comment:
this gets run everytime a tag is selected within the metric selector. intentional?

Reply:
also, since these are all derived values, wouldn't it make more sense to use a `useMemo` here?